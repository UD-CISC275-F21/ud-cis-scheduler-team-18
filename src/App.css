

body{
  text-align: center;
  background-color: rgb(72, 71, 175);
}

input{
  background-color: #b5b0b9;
  margin-top: 10px;
}
textarea{
background-color: #b5b0b9;
margin-top: 10px;
}
button{  
color: rgb(179, 175, 175);
background-color: rgb(49, 47, 47);
margin-top: 1%;
margin-bottom: 1%;
width: 15%;
}
td {
  border: 1px solid #333;
}

thead,
tfoot {
  background-color: #333;
  color: #fff;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
}
table{
margin-left: auto;
margin-right: auto;
border: 1px solid #333;
}

#semesterBox{
margin-left: auto;
margin-right: auto;
width: 1000px;
height: auto;
border: 10px solid #333;
padding: 50px;

}
th{
  margin-left: auto;
  margin-right: auto;
}
#courseAdd {
  margin-left: auto;
  margin-right: auto;
  width: auto;
  margin-top: auto;
  margin-bottom: auto;
}