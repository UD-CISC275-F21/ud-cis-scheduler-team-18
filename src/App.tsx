import React from "react";
import "./App.css";
//change Andrew's Branch
function App (): JSX.Element {
	let numSemesters = 0;
	function addSem() {
		numSemesters++;
		const sem = "Semester " + numSemesters.toString();
		const semBox = document.getElementById("semesterBox")!;
		semBox.innerHTML += "<table class = \"sem\"> <h5>" + sem +"</h5></table>";
		//alert(semBox.innerHTML);
	}

	return (
		<body>
			<h1>UD CIS Scheduler Team 18: Andrew McCracken, Luis Garcia</h1>
			<br></br>
			<         div>TODO: Dynamically fill class List. Fill in Semester Planner.</div>
			<h3 id="major">Major:</h3>
			<table id="courseTable">
				<thead>
					<tr>
						<th>Classes Remaining:</th>
					</tr>
				</thead>
				<tbody>
					<tr>
						<td>Class A</td>
						<td>Description</td>
					</tr>
					<tr>
						<td>Class B</td>
						<td>Description</td>
					</tr>
				</tbody>
			</table>

			<h3>Semester Planner:</h3>
			
			<table className = "semesterBox">
			</table>
			<button id="addSem" onClick= {addSem} >Add Semester</button>
			
		
			
		
		</body>
	);
  
}

export default App;
