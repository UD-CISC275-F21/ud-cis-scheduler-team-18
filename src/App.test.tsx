import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";
import renderer from "react-test-renderer";
//basic screen text rendering tests:
test("renders UD CIS Scheduler text", () => {
    render(<App />);
    const linkElement = screen.getByText(/UD CIS Scheduler/i);
    expect(linkElement).toBeInTheDocument();
});

test("renders Course List", () => {
	render(<App />);
    const linkElement = screen.getByText(/Course List:/i);
    expect(linkElement).toBeInTheDocument();
});

test("renders Fulfilled Requirements and unfulfilled requirements", () => {
	render(<App />);
    const linkElement = screen.getAllByText(/Fulfilled Requirements:/i);
    expect(linkElement[0]).toBeInTheDocument();
	expect(linkElement[1]).toBeInTheDocument();
});

test("renders Control Panel", () => {
	render(<App />);
    const linkElement = screen.getByText(/Control Panel/i);
    expect(linkElement).toBeInTheDocument();
});

test("Class Select", () => {
	render(<App />);
    const linkElement = screen.getByText(/Class Select/i);
    expect(linkElement).toBeInTheDocument();
});

test("renders Semester Viewer", () => {
	render(<App />);
    const linkElement = screen.getByText(/Semesters:/i);
    expect(linkElement).toBeInTheDocument();
});

//more tests:

test("renders all 11 buttons for the control panel", () =>{
	render(<App />);
	const linkElement = document.querySelectorAll("Button");
	expect(linkElement).toHaveLength(11);
	expect(linkElement[0]).toHaveTextContent("Previous Class");
	expect(linkElement[1]).toHaveTextContent("Add Class");
	expect(linkElement[2]).toHaveTextContent("Remove Class");
	expect(linkElement[3]).toHaveTextContent("Next Class");
	expect(linkElement[4]).toHaveTextContent("Previous Semester");
	expect(linkElement[5]).toHaveTextContent("Add Semester");
	expect(linkElement[6]).toHaveTextContent("Clear Semester");
	expect(linkElement[7]).toHaveTextContent("Next Semester");
	expect(linkElement[8]).toHaveTextContent("Rename Semester");
	expect(linkElement[9]).toHaveTextContent("Remove Semester");
	expect(linkElement[10]).toHaveTextContent("Reset Semester");
});

test("renders 3 listGroups that are horizontal", () => {
	render(<App />);
	const linkElement = document.querySelectorAll("div");
	let listGroups = 0;
	for(const x in linkElement){
		if(linkElement[x].className == "justify-content-center list-group list-group-horizontal"){
			listGroups++;
		}
	}
	expect(listGroups).toEqual(3);
});

test("renders 1 normal listGroup", () => {
	render(<App />);
	const linkElement = document.querySelectorAll("div");
	let listGroups = 0;
	for(const x in linkElement){
		if(linkElement[x].className == "list-group"){
			listGroups++;
		}
	}
	expect(listGroups).toEqual(1);
});

test("add class button and remove class button", () => {
    render(<App />);
	let listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(8);
    const button = screen.getByText(/Add Class/i);
	button.click();
	listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(9);
	const button2 = screen.getByText(/Remove Class/i);
	button2.click();
	listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(8);
});

test("testing clear semester button", () => {
	render(<App />);
	let listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(8);
	const addButton = screen.getByText(/Add Class/i);
	const nextButton = screen.getByText(/Next Class/i);
	addButton.click();
	nextButton.click();
	addButton.click();
	nextButton.click();
	addButton.click();
	nextButton.click();
	addButton.click();
	nextButton.click();
	listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(12);
	const clearSemButton = screen.getByText(/Clear Semester/i);
	clearSemButton.click();
	listItems = document.getElementsByClassName("list-group-item list-group-item-dark");
	expect(listItems).toHaveLength(8);
});

test("snapshot testing", () => {
	const tree = renderer.create(<App />).toJSON();
	expect(tree).toMatchSnapshot();
	
});