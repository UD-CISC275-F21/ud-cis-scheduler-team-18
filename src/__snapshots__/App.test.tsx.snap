// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<div
  className="App container"
>
  <div
    className="row"
  >
    <h1>
      UD CIS Scheduler Team 18 Andrew McCracken
    </h1>
    <div
      className="row"
    >
      <h1>
        Course List:
      </h1>
      <div
        className="justify-content-center list-group list-group-horizontal"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc275
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc320
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc372
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cpeg470
        </div>
      </div>
      <h2>
        Unfulfilled Requirements:
      </h2>
      <div
        className="justify-content-center list-group list-group-horizontal"
        onKeyDown={[Function]}
      >
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc275
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc320
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cisc372
        </div>
        <div
          className="list-group-item list-group-item-dark"
          data-rr-ui-event-key={null}
          onClick={[Function]}
        >
          cpeg470
        </div>
      </div>
      <h2>
        Fulfilled Requirements:
      </h2>
      <div
        className="justify-content-center list-group list-group-horizontal"
        onKeyDown={[Function]}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h2>
        Control Panel
      </h2>
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Previous Class
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add Class
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Remove Class
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Next Class
        </button>
      </div>
      <br />
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Previous Semester
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add Semester 
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Clear Semester
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Next Semester
        </button>
      </div>
      <br />
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Rename Semester
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Remove Semester
        </button>
        <button
          className="btn btn-secondary btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Reset Semesters
        </button>
      </div>
    </div>
    <div
      className="col"
    >
      <h2>
        Class Select
      </h2>
      <div>
        Class: 
        cisc275
      </div>
      <div>
        Professor: 
        Dr. Bart
      </div>
      <div>
        Prereq: 
        none
      </div>
      <div>
        Credits: 
        3
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <h2>
      Semesters:
    </h2>
    <h5>
      Total Credits: 
      0
    </h5>
    <h6>
      New Semester
    </h6>
    <div
      className="list-group"
      onKeyDown={[Function]}
    />
  </div>
</div>
`;
